langchain_postgres/vectorstores.py

```diff
@@ -701,9 +701,15 @@ def _handle_field_filter(
             queried_field = self.EmbeddingStore.cmetadata[field].astext
 
             if operator in {"$in"}:
-                return queried_field.in_([str(val) for val in filter_value])
+                return func.jsonb_exists_any(
```

Comment:
This change will result in a bug when working with integer fields which is a more common use case.

Adding a unit-test in a separate PR and will rebase.